crates/core/tedge_api/src/entity_store.rs

```diff
@@ -168,13 +247,10 @@ impl EntityStore {
             affected_entities.push(parent.clone());
         }
 
-        let entity_id = message
-            .entity_id
-            .unwrap_or_else(|| self.derive_entity_id(&message.topic_id));
```

Comment:
It was working because of the way the `external_id` was computed and passed into the entity store on an `update`. That compute logic in the converter was using that `@id` as-is when one is provided. But, this is going to to be restored once the computation logic is passed in as a closure.